x/accounts/defaults/lockup/lockup.go

```diff
@@ -416,6 +470,81 @@ func getStakingDenom(ctx context.Context) (string, error) {
 	return resp.Params.BondDenom, nil
 }
 
+// iterates through all the unbonding entries and check if any of the entries are matured and handled.
+func (bva *BaseLockup) CheckUbdEntriesMature(ctx context.Context) error {
+	whoami := accountstd.Whoami(ctx)
+	delAddr, err := bva.addressCodec.BytesToString(whoami)
+	if err != nil {
+		return err
+	}
+
+	currentTime := bva.headerService.HeaderInfo(ctx).Time
+
+	removeKeys := []string{}
+	err = bva.UnbondEntries.Walk(ctx, nil, func(key string, value lockuptypes.UnbondingEntries) (stop bool, err error) {
+		for i := 0; i < len(value.Entries); i++ {
+			entry := value.Entries[i]
+			// if not mature then skip
+			if entry.EndTime.After(currentTime) {
+				return false, nil
+			}
+
+			skipIteration := false
+			entries, err := bva.getUbdEntries(ctx, delAddr, key)
+			if err != nil {
+				// if ubd delegation is empty then skip the next iteration check
+				if !errorsmod.IsOf(err, stakingtypes.ErrNoUnbondingDelegation) {
+					return true, err
+				}
+
+				skipIteration = true
+			}
+
+			found := false
+			// check if the entry is still exist in the unbonding entries
+			if !skipIteration {
+				for _, e := range entries {
+					if e.CompletionTime.Equal(entry.EndTime) && entry.CreationHeight == entry.CreationHeight {
```

Comment:
hmmm not sure i understand the question here. But this seems to be the only way to check if entries is actually removed from staking ( AKA is handled ) since entry can still be put onhold after matured time